pytorch_lightning/plugins/collective/collective_plugin.py

```diff
@@ -0,0 +1,56 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional, Union
+
+import torch
+
+from pytorch_lightning.utilities.distributed import ReduceOp
+
+
+class Collective(ABC):
+    """Base class for collective functions for training type plugins."""
+
+    @abstractmethod
+    def barrier(self, name: Optional[str] = None) -> None:
```

Comment:
Named barrier/rendezvous is not supported by all distributed backends we are encapsulating. Maybe `name` shouldn't be an argument in the interface?

Reply:
will *args: Any, **kwargs: Any be better then?

Reply:
I think it's better to not expose the functionality. IMHO the value of the interface is to offer a consistent semantic across different underlying implementation, so ideally feature wise it captures the lowest common denominator.

Reply:
Yeah, the only place needs name param in barrier is TPU, TPU barrier() is calling [rendezvous]( ()https://pytorch.org/xla/release/1.5/_modules/torch_xla/core/xla_model.html). But base on [LSP](https://en.wikipedia.org/wiki/Liskov_substitution_principle) we needs include it in the base collective api. Will add comments as you suggested "For backends that don't support named barrier, the argument is ignores and only serves as an annotation"